src/Rendering/prePassRenderer.ts

```diff
@@ -0,0 +1,280 @@
+import { PBRBaseMaterial } from "../Materials/PBR/pbrBaseMaterial";
+import { MultiRenderTarget } from "../Materials/Textures/multiRenderTarget";
+import { Scene } from "../scene";
+import { Engine } from "../Engines/engine";
+import { Constants } from "../Engines/constants";
+import { ImageProcessingPostProcess } from "../PostProcesses/imageProcessingPostProcess";
+import { SubSurfaceScatteringPostProcess } from "../PostProcesses/subSurfaceScatteringPostProcess";
+import { Effect } from "../Materials/effect";
+import { _DevTools } from '../Misc/devTools';
+import { Color4 } from "../Maths/math.color";
+import { SubSurfaceConfiguration } from "./subSurfaceConfiguration";
+
+/**
+ * Renders a pre pass of the scene
+ * This means every mesh in the scene will be rendered to a render target texture
+ * And then this texture will be composited to the rendering canvas with post processes
+ * It is necessary for effects like subsurface scattering or deferred shading
+ */
+export class PrePassRenderer {
+    /** @hidden */
+    public static _SceneComponentInitialization: (scene: Scene) => void = (_) => {
+        throw _DevTools.WarnImport("PrePassRendererSceneComponent");
+    }
+
+    private _scene: Scene;
+    private _engine: Engine;
+    private _isDirty: boolean = false;
+
+    /**
+     * Number of textures in the multi render target texture where the scene is directly rendered
+     */
+    public readonly mrtCount: number = 4;
+
+    /**
+     * The render target where the scene is directly rendered
+     */
+    public prePassRT: MultiRenderTarget;
+    private _mrtTypes = [
+        Constants.TEXTURETYPE_HALF_FLOAT, // Original color
+        Constants.TEXTURETYPE_HALF_FLOAT, // Irradiance
+        Constants.TEXTURETYPE_HALF_FLOAT, // Depth (world units)
+        Constants.TEXTURETYPE_UNSIGNED_INT // Albedo
+    ];
+    private _multiRenderAttachments: number[];
+    private _defaultAttachments: number[];
+    private _clearAttachments: number[];
+
+    private readonly _clearColor = new Color4(0, 0, 0, 0);
+
+    /**
+     * Defines the ratio real world => scene units.
+     * Used for subsurface scattering
+     */
+    public metersPerUnit: number = 1;

```

Comment:
this should be in the subsurface configuration ?

Reply:
it should :)